temporalio/client.py

```diff
@@ -0,0 +1,935 @@
+                args=args,
+                id=id,
+                task_queue=task_queue,
+                execution_timeout=execution_timeout,
+                run_timeout=run_timeout,
+                task_timeout=task_timeout,
+                id_reuse_policy=id_reuse_policy,
+                retry_policy=retry_policy,
+                cron_schedule=cron_schedule,
+                memo=memo,
+                search_attributes=search_attributes,
+                header=header,
+                start_signal=start_signal,
+                start_signal_args=start_signal_args,
+            )
+        )
+
+    async def execute_workflow(
+        self,
+        workflow: str,
+        *args: Any,
+        id: str,
+        task_queue: str,
+        execution_timeout: Optional[timedelta] = None,
+        run_timeout: Optional[timedelta] = None,
+        task_timeout: Optional[timedelta] = None,
+        id_reuse_policy: WorkflowIDReusePolicy = WorkflowIDReusePolicy.ALLOW_DUPLICATE,
+        retry_policy: Optional[temporalio.common.RetryPolicy] = None,
+        cron_schedule: str = "",
+        memo: Optional[Mapping[str, Any]] = None,
+        search_attributes: Optional[Mapping[str, Any]] = None,
+        header: Optional[Mapping[str, Any]] = None,
+        start_signal: Optional[str] = None,
+        start_signal_args: Iterable[Any] = [],
+    ) -> Any:
+        """Start a workflow and wait for completion.
+
+        This is a shortcut for :py:meth:`start_workflow` +
+        :py:meth:`WorkflowHandle.result`.
+        """
+        return await (
+            await self.start_workflow(
+                workflow,
+                *args,
+                task_queue=task_queue,
+                id=id,
+                execution_timeout=execution_timeout,
+                run_timeout=run_timeout,
+                task_timeout=task_timeout,
+                id_reuse_policy=id_reuse_policy,
+                retry_policy=retry_policy,
+                cron_schedule=cron_schedule,
+                memo=memo,
+                search_attributes=search_attributes,
+                header=header,
+                start_signal=start_signal,
+                start_signal_args=start_signal_args,
+            )
+        ).result()
+
+    def get_workflow_handle(
+        self, workflow_id: str, *, run_id: Optional[str] = None
+    ) -> "WorkflowHandle[Any]":
+        """Get a workflow handle to an existing workflow by its ID.
+
+        Args:
+            workflow_id: Workflow ID to get a handle to.
+            run_id: Run ID that will be used for all calls.
+        """
+        return WorkflowHandle(self, workflow_id, run_id=run_id)
+
+
+T = TypeVar("T")
+
+
+class WorkflowHandle(Generic[T]):
+    """Handle for interacting with a workflow.
+
+    This is usually created via :py:meth:`Client.get_workflow_handle` or
+    returned from :py:meth:`Client.start_workflow`/:py:meth:`Client.execute_workflow`.
+    """
+
+    SELF_RUN_ID = "_<self_run_id>_"
```

Comment:
You could use `object()` here instead to create a unique value but it'd probably conflict with the type definitions.

Reply:
I can make the type definitions use a literal actually, e.g. `Union[None, str, typing_extensions.Literal["_<self_run_id>_"]]` or maybe just `Union[None, str, Type[SELF_RUN_ID]]`. I'll investigate.

Reply:
This no longer applies as people are not allowed to provide run IDs as parameters to calls